cli_tools/gce_vm_image_import/importer/importer.go

```diff
@@ -246,6 +248,19 @@ func Run(clientID string, imageName string, dataDisk bool, osID string, customTr
 
 		daisyutils.PostProcessDErrorForNetworkFlag("image import", err, network, w)
 
+		// When translation fails, report the detection results if they don't
+		// match the user's input.
+		fromUser, _ := distro.ParseGcloudOsParam(osID)
+		detected, _ := distro.FromLibguestfs(
+			w.GetSerialConsoleOutputValue("detected_distro"),
+			w.GetSerialConsoleOutputValue("detected_major_version"),
+			w.GetSerialConsoleOutputValue("detected_minor_version"))
+		if fromUser != nil && detected != nil && !fromUser.ImportCompatible(detected) {
+			// The error is already logged by Daisy, so skipping re-logging it here.
+			return w, fmt.Errorf("%q was detected on your disk, "+
+				"but %q was specified. Please verify and re-import",
+				detected.AsGcloudArg(), fromUser.AsGcloudArg())
+		}
```

Comment:
1. This appears to indiscriminately hides the original error when detected log values are written. What if import returned detected OS version import failed because another reason? 
2. Even if the cause is mismatched OS flag value, for debugging and sanity checks, we should include the original error as well. 
3. This code should be extracted to a func to reduce the size of Run() and improve it's readability. Ideally to a separate util file.

Reply:
talked offline